File: org.argeo.app.geo/src/org/argeo/app/geo/ux/MapPart.java
package org.argeo.app.geo.ux;

/** An UX part displaying a map. */
public interface MapPart {
	/** A supported geographical data format. */
	enum GeoFormat {
		GEOJSON, GPX;
	}

	void addPoint(double lng, double lat, String style);

	void addUrlLayer(String url, GeoFormat format, String style);

	void setZoom(int zoom);

	void setCenter(double lng, double lat);

	/** Event when a feature has been single-clicked. */
	record FeatureSingleClickEvent(String path) {
	};

	/** Event when a feature has been selected. */
	record FeatureSelectedEvent(String path) {
	};

	/** Event when a feature popup is requested. */
	record FeaturePopupEvent(String path) {
	};
}

File: org.argeo.app.geo.js/src/org.argeo.app.geo.js/MapPart.js
/** API to be used by Java.
 *  @module MapPart
 */

/** Abstract base class for displaying a map. */
export default class MapPart {

	/** Zoom the map to the given value. */
	setZoom(zoom) {
		throw new Error("Abstract method");
	}

	/** Set the center of the map to the given coordinates. */
	setCenter(lng, lat) {
		throw new Error("Abstract method");
	}
}
